roles/nameserver/README.rst

```diff
@@ -0,0 +1,170 @@
+- Manages forward and reverse DNS records
+
+Prerequisites
++++++++++++++
+
+- CentOS 7.2
+
+Variables
++++++++++
+Most variables are defined in ``roles/nameserver/defaults/main.yml`` and values are chosen to support our Sepia_ lab.  They can be overridden in the ``secrets`` repo.
+
+A list of packages to install that is specific to the role.  The list is defined in ``roles/nameserver/vars/packages.yml``::
+
+    packages: []
+
+BIND main configuration directory.  Defined in ``roles/nameserver/defaults/main.yml``::
+
+    named_conf_dir: "/var/named"
+
+BIND main configuration file.  This is the default CentOS path.  Defined in ``roles/nameserver/defaults/main.yml``::
+
+    named_conf_file: "/etc/named.conf"
+
+BIND data directory.  named debug output and statistics are stored here.  Defined in ``roles/nameserver/defaults/main.yml``::
+
+    named_conf_data_dir: "/var/named/data"
+
+Port BIND should listen on.  Defined in ``roles/nameserver/defaults/main.yml``::
+
+    named_conf_listen_port: 53
+
+Interface(s) BIND should listen on.  This defaults to listen on all IPv4 interfaces Ansible detects for the nameserver.  Defined in ``roles/nameserver/defaults/main.yml``::
+
+    named_conf_listen_iface:
+      - 127.0.0.1
+      - "{{ ansible_all_ipv4_addresses[0] }}"
+
+Path to BIND zone files.  Defined in ``roles/nameserver/defaults/main.yml``::
+
+    named_conf_zones_path: "/var/named/zones"
+
+named daemon options.  Writes to ``/etc/sysconfig/named``.  Defined in ``roles/nameserver/defaults/main.yml``::
+
+    named_conf_daemon_opts: []
+
+    # Example for IPv4 support only:
+    named_conf_daemon_opts: "-4"
+
+The following variables are self-explanatory and written to each zone file's SOA resource record.  They're defined in ``roles/nameserver/defaults/main.yml``::
+
+    named_conf_ttl: 3600
+    named_conf_refresh: 3600
+    named_conf_retry: 3600
+    named_conf_expire: 604800
+
+Desired primary nameserver and admin e-mail for each zone file.  Defined in the secrets repo::
+
+    named_conf_soa: []
+
+    # Example:
+    named_conf_soa: "ns1.example.com. admin.example.com."
+
+----
+
+**named_domains**
+
+The ``named_domains`` dictionary is the bread and butter of creating zone files.  It is in standard YAML syntax.  Each domain (key) must have ``forward`` and ``ipvar`` defined although ``ipvar`` can be set to ``NULL``.  Optional values include ``miscrecords`` and ``reverse``.
+
+``forward``
+  The domain of the forward lookup zone for each domain (key)
+
+``ipvar``
+  The variable assigned to a system in the Ansible inventory.  This allows systems to have multiple IPs assigned for a front and ipmi network, for example.  See **Inventory Example** below.
+
+``miscrecords``
+  Records to add to corresponding ``forward`` zone file.  This is a good place for CNAMEs and MX records and records for hosts you don't have in your Ansible inventory.  If your main nameserver is in a subdomain, you should create a record here.  See example.
```

Comment:
"If your main nameserver is in a subdomain, you should create a record here"...does this mean the glue record, or something else?